geps/gep-2643/index.md

```diff
@@ -0,0 +1,235 @@
+    protocol: TLS
+    hostname: "*.example.com"
+    allowedRoutes:
+      namespaces:
+        from: Same
+      kinds:
+      - kind: TLSRoute
+    tls:
+      mode: Passthrough
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: TLSRoute
+metadata:
+  name: my-tls-route
+spec:
+  parentRefs:
+  - name: gateway-tlsroute
+  hostnames:
+  - abc.example.com
+  rules:
+  - backendRefs:
+    - name: tls-backend
+      port: 443
+```
+
+A typical [north/south](https://gateway-api.sigs.k8s.io/concepts/glossary/#northsouth-traffic)
+API request flow for a gateway implemented using a `TLSRoute` is:
+* A client makes a request to https://foo.example.com.
+* DNS resolves the name to a `Gateway` address.
+* The reverse proxy receives the request on a `Listener` and uses the
+[Server Name Indication](https://datatracker.ietf.org/doc/html/rfc6066#section-3)
+attribute to match an `TLSRoute`.
+* The reverse proxy passes through the request directly to one or more objects,
+i.e. `Service`, in the cluster based on `backendRefs` rules of the `TLSRoute`.
+
+## Conflict management and precedences
+
+The following conflict situations are covered by TLSRoute and TLS passthrough cases:
+
+* When a Gateway contains a listener with `protocol=TLS`, the Gateway MUST NOT
+allow any other kind of listener on the same port and the Gateway SHOULD be
+marked as `Accepted=False` in case there are different kinds of listeners on
+the same port.  
+* When a Gateway contains a listener with `protocol=TLS` and `tls.mode=Passthrough`,
+the `Gateway` MUST NOT allow another listener on the same port with a different
+`tls.mode` and the `Gateway` SHOULD be marked as `Accepted=False`.  
+* Any violating Listener should have a Condition `Conflicted=True`.  
+* If a hostname is specified by both the `Listener` and `TLSRoute`, there must
+be at least one intersecting hostname for the `TLSRoute` to be attached to the
+`Listener`.  
+  * A `Gateway listener` with `test.example.com` as the hostname matches a `TLSRoute` that
+  specifies `test.example.com` but does not match a `TLSRoute` that specifies `*.example.com`  
+  * A `Gateway listener` with `*.example.com` as the hostname matches a `TLSRoute` that
+  specifies at least one hostname that matches the `Gateway listener` hostname.
+  For example, `test.example.com` and `*.example.com` would both match. On the
+  other hand, `example.com` and `test.example.net` would not match.  
+  * If both the `Gateway listener` and `TLSRoute` specify hostnames, any `TLSRoute`
+  hostnames that do not match the `Gateway listener` hostname MUST be ignored
+  for that Listener. For example, if a `Gateway listener` specified `*.example.com`,
+  and the  `TLSRoute` specified `test.example.com` and `test.example.net`,
+  the later must not be considered for a match.  
+  * In any of the cases above, the `TLSRoute` should have a `Condition` of `Accepted=True`.
+
+## Conformance Details
+
+###  Feature Names
+
+* TLSRoute
+
+### Conformance tests
```

Comment:
cc @rostislavbobo so we can discuss a bit more about it :)

Reply:
We'll definitely need conformance tests for
- Mixed TLS termination on Gateway listeners, which today will mark all such listeners as `Conflicted` with `Reason: ListenerConflict` due to incompatible TLS modes (see [comment](https://github.com/kubernetes-sigs/gateway-api/pull/4064/files#r2512417073))
- Mixed TLS termination on Gateway listener and ListenerSet listener, where the conflicting listener on the Gateway is accepted based on the [Listener Precedence](https://gateway-api.sigs.k8s.io/geps/gep-1713/#listener-precedence)

Reply:
ok, closing this thread then. We should map this on ListenerSet gep